classes/local/util/manager.php

```diff
@@ -67,9 +67,42 @@ public static function instance($logger) {
     }
 
     /**
+     * Returns a list of (component => name) values.
+     *
+     * @return string[].
+     */
+    public static function get_plugintype_names() {
+
+        $pluginman = \core_plugin_manager::instance();
+        $plugintypes = $pluginman->get_plugin_types();
+
+        // Replacing the directory with the plugin name.
+        foreach ($plugintypes as $type => $dir) {
+            $plugintypes[$type] = $pluginman->plugintype_name($type);
+        }
+
+        return $plugintypes;
+    }
+
+    /**
+     * Returns a list of variables needed by the plugin templates.
+     *
+     * @param string $component.
+     * @return string[].
+     */
+    public static function get_component_variables($component) {
```

Comment:
This is not complete yet, I'll add code here as I write the get_template_variables() function for the skel classes.

For each skel class that represents a template which requires variables to be present in the recipe there'll be a call like this one:
```php
$recipevariables = \tool_pluginskel\local\skel\skelclass::get_template_variables($component);
```